### PR TITLE
✨Source Recharge: Migrate Python CDK to Low-code CDK

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
*Migrating Source Recharge to Low-Code CDK*
Closes airbytehq/airbyte-internal-issues#2001

## How
*Developed using (Configuration Based Source) low-code CDK*

## Recommended reading order
1. `spec.yaml`
2. `manifest.yaml`
3. `schemas/*`

## 🚨 User Impact 🚨
- No breaking changes, migration to low-code
<details><summary><strong>Migrate Connector: Source Recharge</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@btkcodedev can you share the output from integration tests?

**Comment by btkcodedev** [comment]
Can't get the credentials due to premium plans requirement.
Still used invalid creds to match with tests
Syntax and checks until the connection with schema are all passed.

image

**Comment by btkcodedev** [comment]
Incremental sync, pagination, and base stream all migrated from python to low code individually. 
Waiting for comments

**Comment by marcosmarxm** [comment]
> Can't get the credentials due to premium plans requirement. Still used invalid creds to match with tests Syntax and checks until the connection with schema are all passed.
> 
> image

@RealChrisSean can you share Airbyte's test account so they can implement properly the migration?

**Comment by btkcodedev** [comment]
@RealChrisSean Any update with these?

**Comment by RealChrisSean** [comment]
Still working on getting credentials. We have a back log at the moment.

**Comment by btkcodedev** [comment]
@sh4sh It would be very helpful if you run the test for these with airbyte-recharge credentials and review any changes that are required.
Thanks :)

File: airbyte-integrations/connectors/source-recharge/source_recharge/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      title: Access Token
+      description: The value of the Access Token generated. See the <a href="https://docs.airbyte.com/integrations/sources/recharge">docs</a>
+        for more information.
+      airbyte_secret: true
+    owner_resource:
+      type: string
+      title: Owner Resource value for metafields resource
+      description: The value will be passed as request option. Possible values include - store, customer or subscription. Default is store, See the <a href="https://developer.rechargepayments.com/2021-01/metafields/metafields_list">docs</a>
+        for more information.
+      default: store
```
Comment (sh4sh) [inline airbyte-integrations/connectors/source-recharge/source_recharge/spec.yaml:32]: In the Python connector the `owner_resource` value is not sourced through connector configuration, it is inferred using stream slices/partitions like below. We should implement it the same way here. https://github.com/airbytehq/airbyte/blob/b19ed6d8b398f09b422a888a5a043b964a0a21f9/airbyte-integrations/connectors/source-recharge/source_recharge/api.py#L147-L165

**Comment by btkcodedev** [comment]
@sh4sh I've tried to implement slice according to my understanding, I am not able to test it as I don't have any creds./
Please check :)

**Comment by marcosmarxm** [comment]
Hello, and thank you so much for contributing to the project during the Hackathon. Due to the high complexity of the code migration, we still have a few contributions open. I'm sending this message to update you that this contribution will be added to the final points we're going to announce tomorrow. With that said, we ask for your help to continue the process and finish the review.

**Comment by btkcodedev** [comment]
Thank you @marcosmarxm

**Pull request closed without merging**